pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -0,0 +1,130 @@
+    sqldb.query(sql.assessment_list, params, function(err, result) {
+        if (ERR(err, next)) return;
+        res.locals.assessment_list_rows = result.rows;
+        sqldb.query(sql.select_groups, params, function(err, result) {
+            if (ERR(err, next)) return;
+            res.locals.groups_rows = result.rows;
+            sqldb.query(sql.not_assigned_users, params, function(err, result) {
+                if (ERR(err, next)) return;
+                res.locals.not_assigned_users_rows = result.rows;
+                debug('render page');
+                res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+            });
+        });
+    });
+});
+
+router.post('/', function(req, res, next) {
+    if (!res.locals.authz_data.has_instructor_edit) return next();
+    if (req.body.__action == 'upload_assessment_groups') {
+        groupUpdate.uploadInstanceGroups(res.locals.assessment.id, req.file, res.locals.user.user_id, res.locals.authn_user.user_id, function(err, job_sequence_id) {
+            if (ERR(err, next)) return;
+            res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
+        });
+    } else if (req.body.__action == 'auto_assessment_groups') {
+        groupUpdate.autoGroups(res.locals.assessment.id, res.locals.user.user_id, res.locals.authn_user.user_id, req.body.maxnum, req.body.minnum, req.body.optradio, function(err, job_sequence_id) {
+            if (ERR(err, next)) return;
+            res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
+        });
+    } else if (req.body.__action == 'copy_assessment_groups') {
+        const params = [
+            res.locals.assessment.id,
+            req.body.inputGroupSelect01,
+        ];
+        sqldb.call('assessment_groups_copy', params, function(err, _result) {
+            if (ERR(err, next)) return;
+            res.redirect(req.originalUrl);
+        });
+    } else if (req.body.__action == 'delete_all') {
+        const params = [
+            res.locals.assessment.id,
+            res.locals.authn_user.user_id,
+        ];
+        sqldb.call('assessment_groups_delete_all', params, function(err, _result) {
+            if (ERR(err, next)) return;
+            res.redirect(req.originalUrl);
+        });
+    } else if (req.body.__action == 'addGroup') {
+        const assessment_id = res.locals.assessment.id;
+        const groupname = req.body.groupname;
+        const uids = req.body.uids;
+        const uidlist = uids.split(/[ ,]+/);
+        (async () => {
+            for(const uid of uidlist){
+                if(uid != ' '){
+                    let params = [
+                        assessment_id,
+                        groupname,
+                        uid,
+                    ];
+                    await sqldb.callAsync('assessment_groups_update', params);
+                }
+            }
+        })();
+        res.redirect(req.originalUrl);
```

Comment:
I think we'll want to be careful here because the page redirect could occur before the SQL call finishes.  Maybe you can use the callback version of `sqldb.call()`, and put the redirect in the callback?  Or you could even put the redirect in the async function after the `await sqldb.callAsync()`.